test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -684,7 +684,8 @@ private static void deleteAllILMPolicies() throws IOException {
             Response response = adminClient().performRequest(new Request("GET", "/_ilm/policy"));
             policies = entityAsMap(response);
         } catch (ResponseException e) {
-            if (RestStatus.METHOD_NOT_ALLOWED.getStatus() == e.getResponse().getStatusLine().getStatusCode()) {
+            if (RestStatus.METHOD_NOT_ALLOWED.getStatus() == e.getResponse().getStatusLine().getStatusCode() ||
```

Comment:
Looks like some leftover code to work around test failures?

Reply:
So this was an interesting one; all the REST tests started failing with 400 errors, and it took me a while to track down what was happening.  Previously, if SLM or ILM were not enabled, a call to `GET _ilm/policy` would be interpreted as an index matching `{index}/{type}`, which only has handlers registered for `PUT` and `POST`, and so would return a `406 Method not allowed` error.  Now, the `{index}/{type}` handler has gone away, and so we just get a `400 Bad Request` instead.  I don't think we need to check for `406` errors any more, so I'll try removing that line.

Reply:
Looks like we still need to check for both due to BWC checks

Reply:
Got it, this change makes sense to me.